tests/sources/test_salesforce.py

```diff
@@ -1121,6 +1555,138 @@ async def test_get_all_with_content_docs_and_extraction_service(mock_responses):
+        (
+            "SELECT FIELDS(ALL) FROM Account LIMIT 200 OFFSET 0",
+            "SELECT FIELDS(ALL) FROM Account LIMIT 200 OFFSET 0",
+        ),
+    ],
+)
+async def test_modify_query(soql_query, modified_query):
+    async with create_salesforce_source() as source:
+        query = source.salesforce_client.modify_soql_query(soql_query)
+        assert query == modified_query
+
+
+@pytest.mark.parametrize(
+    "filtering, expected_docs",
+    [
+        (
+            Filter(
+                {
+                    ADVANCED_SNIPPET: {
+                        "value": [
+                            {
+                                "query": "SELECT Id, Name FROM Account",
+                                "language": "SOQL",
+                            }
+                        ]
+                    }
+                }
+            ),
+            [EXPECTED_ACCOUNT_RESPONSE, EXPECTED_CONTENT_RESPONSE],
+        ),
+        (
+            Filter(
+                {
+                    ADVANCED_SNIPPET: {
+                        "value": [
+                            {
+                                "query": "SELECT FIELDS(CUSTOM) FROM Account",
+                                "language": "SOQL",
+                            }
+                        ]
+                    }
+                }
+            ),
+            [EXPECTED_ACCOUNT_RESPONSE, EXPECTED_CONTENT_RESPONSE],
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_docs_for_soql_query(mock_responses, filtering, expected_docs):
+    async with create_salesforce_source() as source:
+        mock_responses.get(
+            TEST_FILE_DOWNLOAD_URL,
+            status=200,
+            body=b"chunk1",
+        )
+        mock_responses.get(
+            TEST_QUERY_MATCH_URL, repeat=True, callback=salesforce_query_callback
+        )
+
+        resultant_docs = []
+        async for record, _ in source.get_docs(filtering):
+            resultant_docs.append(record)
+
+
+@pytest.mark.parametrize(
+    "filtering",
+    [
+        Filter(
+            {
+                ADVANCED_SNIPPET: {
+                    "value": [
+                        {
+                            "query": r"FIND {Alex} in ALL FIELDS",
+                            "language": "SOSL",
+                        }
+                    ]
+                }
+            }
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_docs_for_sosl_query(mock_responses, filtering):
+    async with create_salesforce_source() as source:
+        mock_responses.get(
+            TEST_FILE_DOWNLOAD_URL,
+            status=200,
+            body=b"chunk1",
+        )
+        mock_responses.get(
+            TEST_QUERY_MATCH_URL, status=200, payload=SOSL_RESPONSE_PAYLOAD
+        )
+
+        resultant_docs = []
+        async for record, _ in source.get_docs(filtering):
+            resultant_docs.append(record)
+
+
+@pytest.mark.asyncio
+async def test_remote_validation():
```

Comment:
Can you add a case where remote validation passes?

Reply:
Added a positive test case 👍